decoder/ih264d_sei.c

```diff
@@ -853,12 +853,30 @@ WORD32 ih264d_parse_fgc(dec_bit_stream_t *ps_bitstrm, dec_struct_t *ps_dec,
 
                 for(i = 0; i <= ps_sei->s_sei_fgc_params.au1_num_intensity_intervals_minus1[c]; i++)
                 {
+                    /* Although the fag end of both the NALU and the bitstream buffer */
+                    /* is being parsed, not all FGC SEI symbols would have been */
+                    /* decoded semantically. The code below detects this condition */
+                    if((ps_bitstrm->u4_ofst + 8 + 8) >= ps_bitstrm->u4_max_ofst)
+                    {
+                        return ERROR_INV_SEI_FGC_PARAMS;
+                    }
+
                     ps_sei->s_sei_fgc_params.au1_intensity_interval_lower_bound[c][i] =
                         (UWORD8) ih264d_get_bits_h264(ps_bitstrm, 8);
 
                     ps_sei->s_sei_fgc_params.au1_intensity_interval_upper_bound[c][i] =
                         (UWORD8) ih264d_get_bits_h264(ps_bitstrm, 8);
 
+                    /* Although the fag end of both the NALU and the bitstream buffer */
```

Comment:
each one is not 32 bits. so this is wrong.
please move this inside the loop and check for 32 bits in each iteration.

these two are different because in the second approach we are expecting 32 bits are remaining for the next iteration and if the SEV consumes let us say 2 bits in one iteration, we will only look for 32 for next iteration. In the first approach we would have looked at 64 bits.

also please run compliance tests once this change is done to ensure nothing is broken

Reply:
I run compliance on every MR. But none of the MVC bitstreams have FGC SEI's, so I will not be able to test the change to this function.

Reply:
Since this is used in H264 fgc parsing as well, can you please try decoding h.264 streams with fgc data using avcdec and ensure no errors are seen for a valid clip?

Reply:
Both avcdec and mvcdec are not compliant for a avc bitstream containing a valid FGC SEI. 

I shall write a function to check if `ih264d_sev` will output a valid value for the bits within the offset, and return an error if that value is incorrect. This will account for the variability of the umber of bits parsed by `ih264d_sev`

Reply:
I have added a function that checks for an invalid sev

Reply:
What is the size of FGC NAL where we hit this error?
Is there any way we can ensure that minimum size of the bitstream buffer is large enough to accomodate FGC NALs?

Reply:
The size of the NAL is around 290 KB. The buffer allocated is also exactly the NALU size. But since the NALU has erroneous data, parsing it correctly requires more than 290 KB of buffer. 

The PR for `64286` handled a similar error by re-allocating the dynamic buffer. That code is being hit by this bug and the dynamic buffer size is being increased to 290 KB from 280 KB. But this bug causes the parses to overshoot tis buffer.

Reply:
when allocating the new buffer, can we allocate 64 extra bytes for example and then check for 32 bits before calling SEV in this function?

Reply:
Let me try this

Reply:
This doesn't work either; it causes a compliance failure in avcdec for a valid avc bitstream that contains FGC SEI.

Reply:
Can you try by increasing the allocation by 64 bytes, but not adding any checks before SEV parsing?

Reply:
I will do so

Reply:
Thank you iterating on these suggestions.